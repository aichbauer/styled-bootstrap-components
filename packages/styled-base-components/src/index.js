import { BootstrapBaseCss } from './components';

export { BootstrapBaseCss };
