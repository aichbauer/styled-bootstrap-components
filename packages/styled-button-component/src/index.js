import {
  Button,
  LinkButton,
} from './components/Button';
import { ButtonGroup } from './components/ButtonGroup';
import { ButtonToolbar } from './components/ButtonToolbar';

export {
  Button,
  LinkButton,
  ButtonGroup,
  ButtonToolbar,
};
