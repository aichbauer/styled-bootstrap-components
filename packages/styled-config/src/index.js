import { theme } from './theme';
import {
  border,
  borderRadius,
  boxShadow,
  colors,
  fontFamily,
  fontSize,
  fontWeight,
  height,
  margin,
  padding,
  screenSize,
  width,
} from './utils';

export {
  theme,
  // utils
  border,
  borderRadius,
  boxShadow,
  colors,
  fontFamily,
  fontSize,
  fontWeight,
  height,
  margin,
  padding,
  screenSize,
  width,
};
