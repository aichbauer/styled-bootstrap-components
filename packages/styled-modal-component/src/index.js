import { Modal } from './components/Modal';
import { ModalWrapper } from './components/ModalWrapper';
import { ModalBody } from './components/ModalBody';
import { ModalContent } from './components/ModalContent';
import { ModalDialog } from './components/ModalDialog';
import { ModalFooter } from './components/ModalFooter';
import { ModalHeader } from './components/ModalHeader';
import { ModalTitle } from './components/ModalTitle';
import { ModalBackdrop } from './components/ModalBackdrop';

export {
  Modal,
  ModalWrapper,
  ModalBody,
  ModalContent,
  ModalDialog,
  ModalFooter,
  ModalHeader,
  ModalTitle,
  ModalBackdrop,
};
