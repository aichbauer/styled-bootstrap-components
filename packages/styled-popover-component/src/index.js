import { PopoverWithoutTransition, Popover } from './components/Popover';
import { PopoverArrow } from './components/PopoverArrow';
import { PopoverBody } from './components/PopoverBody';
import { PopoverHeader } from './components/PopoverHeader';

export {
  PopoverWithoutTransition,
  Popover,
  PopoverArrow,
  PopoverBody,
  PopoverHeader,
};
