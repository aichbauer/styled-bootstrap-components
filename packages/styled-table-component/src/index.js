import { Table } from './components/Table';
import { Tr } from './components/Tr';

export {
  Table,
  Tr,
};
