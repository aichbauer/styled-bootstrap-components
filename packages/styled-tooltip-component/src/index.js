import { TooltipWithoutTransition, Tooltip } from './components/Tooltip';
import { TooltipArrow } from './components/TooltipArrow';
import { TooltipInner } from './components/TooltipInner';

export {
  TooltipWithoutTransition,
  Tooltip,
  TooltipArrow,
  TooltipInner,
};
