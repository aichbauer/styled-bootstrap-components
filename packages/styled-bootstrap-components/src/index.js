import { Alert } from 'styled-alert-component';
import { Badge } from 'styled-badge-component';
import {
  A,
  Abbr,
  Address,
  Article,
  Aside,
  B,
  Blockquote,
  Button as BaseButton,
  Caption,
  Code,
  Dd,
  Dfn,
  Dialog,
  Dl,
  Dt,
  Fieldset,
  Figcaption,
  Figure,
  Footer,
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  Header,
  Hgroup,
  Hr,
  Img,
  Input,
  Kbd,
  Legend,
  Label,
  Main,
  Nav as Navigation,
  Ol,
  Optgroup,
  Output,
  P,
  Pre,
  Progress,
  Samp,
  Section,
  Select,
  Small,
  Strong,
  Sub,
  Summary,
  Sup,
  Svg,
  Table as BaseTable,
  Textarea,
  Th,
  Ul,
} from 'styled-base-components';
import {
  Breadcrumb,
  BreadcrumbItem,
} from 'styled-breadcrumb-component';
import {
  Button,
  ButtonGroup,
  ButtonToolbar,
} from 'styled-button-component';
import {
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardImageHeader,
  CardText,
  CardTitle,
} from 'styled-card-component';
import { screenSize } from 'styled-config';
import { Container } from 'styled-container-component';
import {
  Dropdown,
  DropdownItem,
  DropdownMenu,
  DropdownDivider,
} from 'styled-dropdown-component';
import {
  FormCheck,
  FormCheckInput,
  FormControl,
  FormControlFile,
  FormControlPlainText,
  FormGroup,
  FormInline,
  FormText,
  InputGroup,
  InputGroupAppend,
  InputGroupPrepend,
  InputGroupText,
} from 'styled-form-component';
import {
  Column,
  Row,
} from 'styled-grid-system-component';
import { Jumbotron } from 'styled-jumbotron-component';
import {
  ListGroup,
  ListGroupItem,
} from 'styled-listgroup-component';
import {
  Modal,
  ModalBody,
  ModalContent,
  ModalDialog,
  ModalFooter,
  ModalHeader,
  ModalTitle,
} from 'styled-modal-component';
import {
  Nav,
  NavLink,
} from 'styled-nav-component';
import {
  Navbar,
  NavbarLink,
} from 'styled-navbar-component';
import {
  Popover,
  PopoverArrow,
  PopoverBody,
  PopoverHeader,
} from 'styled-popover-component';
import {
  Table,
  Tr,
} from 'styled-table-component';
import {
  Tooltip,
  TooltipArrow,
  TooltipInner,
} from 'styled-tooltip-component';

export {
  Alert,
  Badge,
  A,
  Abbr,
  Address,
  Article,
  Aside,
  B,
  Blockquote,
  BaseButton,
  Caption,
  Code,
  Dd,
  Dfn,
  Dialog,
  Dl,
  Dt,
  Fieldset,
  Figcaption,
  Figure,
  Footer,
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
  Header,
  Hgroup,
  Hr,
  Img,
  Input,
  Kbd,
  Legend,
  Label,
  Main,
  Navigation,
  Ol,
  Optgroup,
  Output,
  P,
  Pre,
  Progress,
  Samp,
  Section,
  Select,
  Small,
  Strong,
  Sub,
  Summary,
  Sup,
  Svg,
  BaseTable,
  Textarea,
  Th,
  Ul,
  Breadcrumb,
  BreadcrumbItem,
  Button,
  ButtonGroup,
  ButtonToolbar,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardImageHeader,
  CardText,
  CardTitle,
  screenSize,
  Container,
  Dropdown,
  DropdownItem,
  DropdownMenu,
  DropdownDivider,
  FormCheck,
  FormCheckInput,
  FormControl,
  FormControlFile,
  FormControlPlainText,
  FormGroup,
  FormInline,
  FormText,
  InputGroup,
  InputGroupAppend,
  InputGroupPrepend,
  InputGroupText,
  Column,
  Row,
  Jumbotron,
  ListGroup,
  ListGroupItem,
  Modal,
  ModalBody,
  ModalContent,
  ModalDialog,
  ModalFooter,
  ModalHeader,
  ModalTitle,
  Nav,
  NavLink,
  Navbar,
  NavbarLink,
  Popover,
  PopoverArrow,
  PopoverBody,
  PopoverHeader,
  Table,
  Tr,
  Tooltip,
  TooltipArrow,
  TooltipInner,
};
