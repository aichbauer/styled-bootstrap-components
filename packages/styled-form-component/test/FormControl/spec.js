import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { FormControl } from '../../src';

test('Styles matches FormControl', () => {
  const tree = renderer.create(<FormControl />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('display', 'block');
  expect(tree).toHaveStyleRule('width', '100%');
  expect(tree).toHaveStyleRule('padding', '0.375rem 0.75rem');
  expect(tree).toHaveStyleRule('font-size', '1rem');
  expect(tree).toHaveStyleRule('line-height', '1.5');
  expect(tree).toHaveStyleRule('color', '#495057');
  expect(tree).toHaveStyleRule('background-color', '#fff');
  expect(tree).toHaveStyleRule('background-clip', 'padding-box');
  expect(tree).toHaveStyleRule('border', '1px solid #ced4da');
  expect(tree).toHaveStyleRule('box-sizing', 'border-box');
  expect(tree).toHaveStyleRule('background-color', 'transparent', {
    modifier: '&::-ms-expand',
  });
  expect(tree).toHaveStyleRule('color', '#495057', {
    modifier: '&:focus',
  });
  expect(tree).toHaveStyleRule('background-color', '#fff', {
    modifier: '&:focus',
  });
  expect(tree).toHaveStyleRule('border-color', '#ced4da', {
    modifier: '&:focus',
  });
  expect(tree).toHaveStyleRule('outline', '0', {
    modifier: '&:focus',
  });
  expect(tree).toHaveStyleRule('box-shadow', '0', {
    modifier: '&:focus',
  });
  expect(tree).toHaveStyleRule('color', '#6c757d', {
    modifier: '&::placeholder',
  });
  expect(tree).toHaveStyleRule('opacity', '1', {
    modifier: '&::placeholder',
  });
});
