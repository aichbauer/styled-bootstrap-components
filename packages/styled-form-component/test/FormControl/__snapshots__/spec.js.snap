// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styles matches FormControl 1`] = `
.c0 {
  display: block;
  width: 100%;
  line-height: 1.5;
  color: #495057;
  background-clip: padding-box;
  -webkit-transition: border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  box-sizing: border-box;
  background-color: #fff;
  border: 1px solid #ced4da;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  border-radius: 0.25rem;
}

.c0::-ms-expand {
  background-color: transparent;
  border: 0;
}

.c0:focus {
  color: #495057;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0,123,255,0.25);
}

.c0::-webkit-input-placeholder {
  color: #6c757d;
  opacity: 1;
}

.c0::-moz-placeholder {
  color: #6c757d;
  opacity: 1;
}

.c0:-ms-input-placeholder {
  color: #6c757d;
  opacity: 1;
}

.c0::placeholder {
  color: #6c757d;
  opacity: 1;
}

.c0:focus {
  background-color: #fff;
}

.c0:focus {
  border-color: #ced4da;
}

.c0:focus {
  box-shadow: 0;
}

<input
  className="c0"
/>
`;
